client/src/alda/Util.java

```diff
@@ -137,6 +146,114 @@ public static String getProgramPath() throws URISyntaxException {
                  .toURI().getPath();
   }
 
+  public static String makeApiCall(String apiRequest) throws IOException {
+      URL url = new URL(apiRequest);
+      HttpURLConnection conn =
+        (HttpURLConnection) url.openConnection();
+
+      if (conn.getResponseCode() != 200) {
+        throw new IOException(conn.getResponseMessage());
+      }
+
+      // Buffer the result into a string
+      BufferedReader rd = new BufferedReader(
+        new InputStreamReader(conn.getInputStream()));
+      StringBuilder sb = new StringBuilder();
+      String line;
+      line = rd.readLine();
+      while (line != null) {
+        sb.append(line);
+        line = rd.readLine();
+      }
+      rd.close();
+      conn.disconnect();
+      return sb.toString();
+  }
+
+  public static void downloadFile(String url, String path) {
+    BufferedInputStream in = null;
+    FileOutputStream fout = null;
+    try {
+      in = new BufferedInputStream(new URL(url).openStream());
+      fout = new FileOutputStream(path);
+
+      final byte data[] = new byte[1024];
+      int count;
+      while ((count = in.read(data, 0, 1024)) != -1) {
+        fout.write(data, 0, count);
+      }
+    } catch (MalformedURLException e) {
+      System.err.println("An error occured while downloading a file (1).");
+      e.printStackTrace();
+    } catch (IOException e) {
+      System.err.println("An error occured while downloading a file (2).");
+      e.printStackTrace();
+    }finally {
+      // Close file IO's
+      try {
+        if (in != null) {
+          in.close();
+        }
+        if (fout != null) {
+          fout.close();
+        }
+      } catch (IOException e) {
+        // We can't do anything.
+        System.err.println("A critical error occured while downoading a file (3).");
+        e.printStackTrace();
+        return;
+      }
+    }
+  }
+
+  public static void updateAlda() throws URISyntaxException {
+    // Get the path to the current alda executable
+    String programPath = getProgramPath();
+    String latestApiStr = "https://api.github.com/repos/alda-lang/alda/releases/latest";
+    String apiResult;
+
+    // Make a call to the Github API to get the latest version number/download URL
+    try {
+      apiResult = makeApiCall(latestApiStr);
+    } catch (IOException e) {
+      System.err.println("There was an error connecting to the Github API.");
+      e.printStackTrace();
+      return;
+    }
+
+    // Turn api result into version numbers and links
+
+    Gson gson = new Gson();
+    JsonObject job = gson.fromJson(apiResult, JsonObject.class);
+
+    // Gets the download URL. This may have ...alda or ...alda.exe
+    String downloadURL = job.getAsJsonArray("assets").get(1).getAsJsonObject().get("browser_download_url").toString();
```

Comment:
It looks like the `get(1)` part of this will always grab the second asset, so we are relying on that being `alda.exe`. I wonder if maybe we should check whether the OS is Windows, then iterate through the assets until we find the download URL that ends with either "alda" or "alda.exe"?